controllers/bpfman-agent/application-program.go

```diff
@@ -78,7 +78,7 @@ func (r *BpfApplicationReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 		for j, p := range a.Spec.Programs {
 			switch p.Type {
 			case bpfmaniov1alpha1.ProgTypeFentry:
-				appProgramId := fmt.Sprintf("%s-%s", strings.ToLower(string(p.Type)), sanitize(p.Fentry.FunctionName))
+				appProgramId := fmt.Sprintf("%s-%s", strings.ToLower(string(p.Type)), p.Fentry.BpfFunctionName)
```

Comment:
I don't think we can make this same change to the types that don't support a list of attach points.  Or, if we do, we should still include the attach point itself, because if a user wants to attach the same bpf program to multiple Fentry attach points (for example), they'll need to use multiple Fentry program entries.

Reply:
Thinking through this more, I think we need both the attach point and the bpf program name.

Reply:
We can get away without the attach point for the programs that support multiple attach points by mandating that people use a single entry in the bpfApplication CRD for a given program-type/bpf-program-name instead of using multiple entries of the same program type for different attach points.